include/inexor/vulkan-renderer/world/collision_query.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/world/collision.hpp"
+
+#include <glm/vec3.hpp>
+
+#include <optional>
+
+// Forward declaration
+namespace inexor::vulkan_renderer::world {
+class Cube;
+} // namespace inexor::vulkan_renderer::world
+
+namespace inexor::vulkan_renderer::world {
+
+// TODO: Implement PointCubeCollision
+
+/// @brief ``True`` of the ray build from the two vectors collides with the cube's bounding box.
+/// @note There is no such function as glm::intersectRayBox.
+/// @param box_bounds An array of two vectors which represent the edges of the bounding box.
+/// @param pos The start position of the ray.
+/// @param dir The direction of the ray.
+/// @return ``True`` if the ray collides with the octree cube's bounding box.
+[[nodiscard]] static bool ray_box_collision(std::array<glm::vec3, 2> box_bounds, glm::vec3 pos, glm::vec3 dir);
```

Comment:
What does the `static` actually do here? The other function doesn't have it, so I'm wondering if it's needed.

Reply:
I think CLion proposed it can be made static.

Reply:
What warning/message does CLion give about it?

Reply:
Now I get it. It's a leftover from when this function was a method. `static` can be removed here :)

Reply:
Done.